src/main/java/io/nats/client/JetStreamManagement.java

```diff
@@ -131,6 +132,29 @@ public interface JetStreamManagement {
      */
     boolean deleteConsumer(String streamName, String consumerName) throws IOException, JetStreamApiException;
 
+    /**
+     * Pauses a consumer.
+     * @param streamName name of the stream
+     * @param consumerName the name of the consumer.
+     * @param pauseUntil consumer is paused until this time.
+     * @return PauseResponse the pause response
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data, for instance the consumer does not exist.
+     */
+    ConsumerPauseResponse pauseConsumer(String streamName, String consumerName, ZonedDateTime pauseUntil) throws IOException, JetStreamApiException;
```

Comment:
I think there should be two api here, one that does not take pauseUntil, since it is not required.

Reply:
Do you mean having three methods in the end:
- `pauseConsumer(streamName, consumerName, pauseUntil)`
- `pauseConsumer(streamName, consumerName)`
- `resumeConsumer(streamName, consumerName)`

When not sending a `pauseUntil`, the consumer does not get paused and should be equivalent to calling `resumeConsumer` instead. So `pauseConsumer` is used to pause a consumer until the set date, and `resumeConsumer` resumes it. Could you explain why you'd expect two api here?

Reply:
Sorry, I thought no pauseUntil meant pause until resume. So just ignore this comment.